frontend/catalyst/passes.py

```diff
@@ -131,56 +158,27 @@ def fn(x):
     will always take precedence over global pass pipelines.
     """
 
-    def _decorator(fn=None, **kwargs):
-        if fn is None:
-            return functools.partial(pipeline, **kwargs)
-
-        if not isinstance(fn, qml.QNode):
-            raise TypeError(f"A QNode is expected, got the classical function {fn}")
-
-        if pass_pipeline is None:
-            # TODO: design a default peephole pipeline
-            return fn
-
-        fn_original_name = fn.__name__
-        wrapped_qnode_function = fn.func
-        fn_clone = copy.copy(fn)
-        fn_clone.__name__ = fn_original_name + "_transformed"
-
-        pass_names = _API_name_to_pass_name()
-
-        def wrapper(*args, **kwrags):
-            # TODO: we should not match pass targets by function name.
-            # The quantum scope work will likely put each qnode into a module
-            # instead of a `func.func ... attributes {qnode}`.
-            # When that is in place, the qnode's module can have a proper attribute
-            # (as opposed to discardable) that records its transform schedule, i.e.
-            #    module_with_transform @name_of_module {
-            #      // transform schedule
-            #    } {
-            #      // contents of the module
-            #    }
-            # This eliminates the need for matching target functions by name.
+    def _decorator(qnode=None):
+        if not isinstance(qnode, qml.QNode):
+            raise TypeError(f"A QNode is expected, got the classical function {qnode}")
```

Comment:
I noticed you removed a pytest that checks for this `TypeError`. Any reason for removing it? Asking since Codecov is flagging this line as not covered by tests.

Reply:
I'll add the test back. I removed the "optional" pipeline and that test was missing a pipeline parameter. But I didn't think about just passing a pipeline there to keep this one. Thanks!

Reply:
https://github.com/PennyLaneAI/catalyst/pull/1349/commits/ed89dd918201ac4ce14b6023c68373e7e8c51dfc